web/component2/src/demos/ref.dart

```diff
@@ -147,149 +146,90 @@ class LogPropsComponent extends UiComponent2<LogPropsProps> {
   }
 }
 
-// ------------ `forwardRef` (deprecated) with Function component ------------
-UiFactory<_LogsPropsFunctionComponentProps> LogsPropsFunctionComponent = forwardRef<_LogsPropsFunctionComponentProps>((props, ref) {
-  return (_LogsPropsFunctionComponent()
+//----------------------------------------------------------------------------//
+// ### Example 2: Exposing inner refs in functional components:
+//----------------------------------------------------------------------------//
+UiFactory<BazProps> LoggingFunctionWrapper = forwardRef<BazProps>((props, ref) {
+  print('LoggingFunctionWrapper rendered!');
+
+  return (Baz()
     ..addProps(props)
+    ..builder = props.builder
     .._forwardedRef = ref)();
-}, displayName: 'LogsPropsFunctionComponent')(_LogsPropsFunctionComponent);
-
-class _LogsPropsFunctionComponentProps = UiProps with LogPropsProps;
+}, displayName: 'LoggingFunctionWrapper')(Baz);
 
-final _LogsPropsFunctionComponent = uiFunction<_LogsPropsFunctionComponentProps>(
-  (props) {
-    final prevPropsRef = useRef<_LogsPropsFunctionComponentProps>(null);
-
-    useEffect(() {
-      if (prevPropsRef.current != null) {
-        print('/////////////////// ${props.demoTitle} /////////////////////');
-        print('old props: ${prevPropsRef.current}');
-        print('new props: $props');
-        print('');
-        print('');
-      }
+mixin BazProps on UiProps {
+  BuilderOnlyUiFactory<UiProps> builder;
 
-      prevPropsRef.current = props;
-    });
+  // Private since we only use this to pass along the value of `ref` to
+  // the return value of forwardRef.
+  //
+  // Consumers can set this private field value using the public `ref` setter.
+  Ref _forwardedRef;
```

Comment:
This shouldn't be typed as `Ref` since it prevents callback refs from being passed in, which is a valid use-case.